rspace/src/test/scala/coop/rchain/rspace/IStoreTests.scala

```diff
@@ -182,6 +183,54 @@ trait IStoreTests
       }
     }
   }
+
+  "pruneHistory" should "work on empty history" in withTestStore { store =>
+    store.pruneHistory(List.empty) shouldBe List.empty
+  }
+
+  it should "return unmodified history when nothing to prune" in withTestStore { store =>
+    forAll("gnat") { (gnat: GNAT[String, Pattern, String, StringsCaptor]) =>
```

Comment:
It would be great if we could test using arbitrarily generated `Seq[GNAT[String, Pattern, String, StringsCaptor]]` or the like, perhaps constraining the choice of channels.

Reply:
Done